packages/flutter/lib/src/services/platform_protocols.dart

```diff
@@ -0,0 +1,219 @@
+  }) async {
+    final dynamic reply = await PlatformMessages.sendJSON(channel, <String, dynamic>{
+      'name': name ?? channel,
+      'args': args,
+    });
+    return _interpretJSON<T>(reply, decoder);
+  }
+
+  /// Creates a broadcast [Stream] for consuming a named platform stream of
+  /// JSON-encoded events.
+  ///
+  /// The optional [args] map must contain valid JSON values according to
+  /// [PlatformMessages.sendJSON]. If [args] is specified, it means that the
+  /// stream is configurable and that multiple streams with different
+  /// configurations might need to co-exist. In that case, [args] must contain
+  /// an `eventChannel` entry with a string value which will be used as the name
+  /// of the channel on which events are emitted for the specified configuration.
+  /// If [args] is left unspecified, the main [channel] itself is used for events.
+  ///
+  /// The platform stream must be implemented on the native side by a handler
+  /// registered for the specified [channel]. That handler should expect calls
+  /// to JSON functions named 'listen' and 'cancel' according to the protocol
+  /// of [invokeJSONFunction]. These functions are used to let the native side
+  /// know when the Dart stream has registered listeners, see
+  /// [StreamController.broadcast]. Both are called with [args] as arguments.
+  ///
+  /// Following the semantics of broadcast streams, `listen` will be called as
+  /// the first listener registers with the returned stream, and `cancel` when
+  /// the last listener cancels its registration. This pattern may repeat
+  /// indefinitely.
+  ///
+  /// If the `listen` invocation fails, en error event is emitted on the
+  /// returned stream. If the `cancel` invocation fails, the error is
+  /// reported through [FlutterError.reportError] as there are no listeners
+  /// to receive an error event in that case.
+  ///
+  /// The native implementation must wrap each data event in an envelope of the
+  /// form
+  ///
+  ///     { "status": "ok",
+  ///       "data": <data>
+  ///     }
+  ///
+  /// Similarly, an error event must be wrapped in an envelope of the form
+  ///
+  ///     { "status": "error",
+  ///       "message": <human-readable error message>,
+  ///       "data": <error details>
+  ///     }
+  ///
+  /// For each successful event, the specified [decoder] is used to turn the
+  /// data into a [T] instance (or `null`), which is then produced as a data
+  /// event by the returned stream. A `null` or missing decoder is treated
+  /// as a constantly `null` function.
+  ///
+  /// On each native error event, the returned stream will produce an error
+  /// event with a [PlatformException] containing the error message and error
+  /// details data.
+  ///
+  /// Malformed JSON or envelopes, or data that the [decoder] cannot handle,
+  /// will make the returned Stream produce an error event containing a
+  /// [FormatException].
+  static Stream<T> createJSONBroadcastStream<T>({
```

Comment:
I would maybe name this as what it does, rather than the implementation behind it. For example, "PlatformProtocols.receiveStream<bool>(channel, { arguments... }, decoder...)".